hypershift-operator/metrics.go

```diff
@@ -103,6 +129,38 @@ func newMetrics(client crclient.Client, log logr.Logger) *hypershiftMetrics {
 			Name: "hypershift_nodepools_size",
 			Help: "Number of replicas associated with a given NodePool",
 		}, []string{"name", "platform"}),
+		// hypershiftOperatorInfo is a metric to capture the current operator details of the management cluster
+		hypershiftOperatorInfo: prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+			Name: HypershiftOperatorInfo,
+			Help: "Number in seconds for the Pod lastTransitionTime to Ready state. Also detailed info about the operator.",
+			ConstLabels: prometheus.Labels{
+				"version":                hypershiftVersion,
+				"image":                  image,
+				"imageId":                imageId,
+				"latestSupportedVersion": latestSupportedVersion,
+				"goVersion":              goVersion,
+				"goArch":                 goArch,
+			},
+		}, func() float64 {
+			ctx := context.TODO()
+			hypershiftNamespace := os.Getenv("MY_NAMESPACE")
+			hypershiftPodName := os.Getenv("MY_NAME")
+			var lastTransitionTime time.Time
+
+			hypershiftPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: hypershiftPodName, Namespace: hypershiftNamespace}}
```

Comment:
Having a second kas request to get this increases the matrix of possible combinations we report unnecessarily e.g. first call succeeds in getOperatoImage so we imageID and this one fail so we report 0.
Let's just set this to 1 always for starters, otherwise I'd expect to get all info from the pod in one single request e.g getOperatoImage -> getOperatorInfo, but we can do that later if we want to let's start simple

Reply:
Ok I will keep on returning 1 to move this forward and add enhancements in the future.